src/standalone/Soil/energy_hydrology.jl

```diff
@@ -1123,3 +1123,55 @@ function soil_turbulent_fluxes_at_a_point(
         dshfdT = dshfdT,
     )
 end
+
+"""
+    ClimaLand.total_liq_water_vol_per_area!(
+        surface_field,
+        model::EnergyHydrology,
+        Y,
+        p,
+        t,
+)
+
+A function which updates `surface_field` in place with the value for
+the total liquid water volume per unit ground area for the `EnergyHydrology`.
+"""
+function ClimaLand.total_liq_water_vol_per_area!(
+    surface_field,
+    model::EnergyHydrology,
+    Y,
+    p,
+    t,
+)
+    earth_param_set = model.parameters.earth_param_set
+    _ρ_liq = LP.ρ_cloud_liq(earth_param_set)
+    _ρ_ice = LP.ρ_cloud_ice(earth_param_set)
+    ClimaCore.Operators.column_integral_definite!(
+        surface_field,
+        Y.soil.ϑ_l .+ Y.soil.θ_i .* _ρ_ice ./ _ρ_liq, # this line allocates
```

Comment:
I think we have subsurface scratch space - we could pass that into this function and use it to store this quantity. But that would likely require changing the function interface, and these functions will only be called to check conservation so they aren't super performance critical

Reply:
We do have access - stored in `p`, but...it's not a default aux variable (only added for some boundary conditions). I could make it a default, and not add it again for other BC

Reply:
Hmm... It would be nice to have it available, independent of the BCs. But I'm happy to keep this PR as-is and address that separately

Reply:
Since this is a relatively large PR already, in terms of number of files changed, I opened an issue and linked in the PR description